helpers/helper-translation-discussion.php

```diff
@@ -190,12 +190,37 @@ function() {
 				'translation_id'     => isset( $this->data['translation_id'] ) ? $this->data['translation_id'] : null,
 				'locale_slug'        => $this->data['locale_slug'],
 				'original_permalink' => $this->data['permalink'],
+				'languages'          => $this->generateCommentFilterLinks( $comments, $this->data['locale_slug'] ),
 			),
 			$this->assets_dir . 'templates'
 		);
 		return $output;
 	}
 
+	public function generateCommentFilterLinks( $comments, $locale_slug ) {
+		$no_of_locale_comments  = 0;
+		$no_of_english_comments = 0;
+		$plus_sign              = '';
+		if ( $comments ) {
+			foreach ( $comments as $comment ) {
+				$comment_locale = get_comment_meta( $comment->comment_ID, 'locale', true );
+				if ( $locale_slug == $comment_locale ) {
+					$no_of_locale_comments++;
+				} else {
+					$no_of_english_comments++;
+					$plus_sign = '+';
+				}
+			}
+		}
+		$gp_locale = GP_Locales::by_slug( $locale_slug );
+		$locale_name = ( $gp_locale ) ? $gp_locale->english_name : 'Current locale';
+
+		return array(
+			$locale_slug => $locale_name . ' (' . $no_of_locale_comments . ')',
+			'all'        => 'Other locales (' . $plus_sign . $no_of_english_comments . ')',
+		);
```

Comment:
This looks like it works but there are a few details that are not quite right, like `$no_of_english_comments` isn't just English comments. An idea for an improved version:

```suggestion
		if ( ! $comments ) {
			// Return early for edge cases.
			return array();
		}

		// Go through all $comments and map their values to the callback of this function.
		$comment_locales = array_map(
			function( $comment ) {
				// Each comment is passed in here and we return the locale.
				return get_comment_meta( $comment->comment_ID, 'locale', true );
			},
			$comments
		);

		// Produce an array of the form $locale => count( entries of array with $locale ).
		$comment_count = array_count_values( $comment_locales );

		// Let's extract the locale count and ensure that it was set.
		$locale_count = isset( $comment_count[ $locale_slug ] ) ? $comment_count[ $locale_slug ] : 0;
		$other_count = array_sum( $comment_count ) - $locale_count;

		$gp_locale = GP_Locales::by_slug( $locale_slug );
		$locale_name = $gp_locale ? $gp_locale->english_name : 'Current locale';

		$comment_links = array(
			$locale_slug => $locale_name . ' (' . $locale_count . ')',
		);

		if ( $other_count ) {
			$comment_links['all'] = 'Other locales (+' . $other_count . ')';
		}

		return $comment_links;
```

With that data in that structure you could also return all languages like this:
![Screenshot 2022-01-19 at 15 22 09](https://user-images.githubusercontent.com/203408/150150649-13853b98-0a7c-4011-ba92-c02a640987c5.png)

```
		return array_filter( array_map( function( $locale_slug, $count ) {
			$locale_name = 'English';
			if ( $locale_slug ) {
				$gp_locale = GP_Locales::by_slug( $locale_slug );
				if ( ! $gp_locale ) {
					return null;
				}
				$locale_name = $gp_locale->english_name;
			}
			return $locale_name . ' (' . $count . ')';
		}, array_keys( $comment_count ), array_values( $comment_count ) ) );
```

Reply:
Thanks for this! Regarding the last sample code block listing all the other locales, that may contradict the change proposed in this issue - https://github.com/Automattic/Orbit/issues/138 - however, we can agree to go with the most appropriate solution.